grant-project/zcash-grant-system
# Request for Proposal (Pt. 3 - Create & Link Proposals)

Another piece to #84.

### What This Does

Another piece of RFPs, which hooks up RFPs and Proposals for user views and admin view. Overall I'm not super thrilled with the UI for a lot of this, but I figured it was better to throw together an MVP than to spin my wheels on this any longer. Here's the full list of changes:

* Added a list of approved proposals on RFPs, and a button to create a proposal linked to one.
    * Proposal create form shows linked proposal, and inherits the title, brief, and category of the RFP (still can be edited.)
* Proposals show their linked RFP on their detail page as well.
* Admin RFPs now list related proposals, and proposals show which RFP (if any) is linked.
* Reworked rfp <-> proposal model to use foreign key instead of lookup table, since it's one-to-many.
* Reworked the way the DraftList component creates new drafts automagically.
* Fixed up proposals create previews to not crash, and look much closer to live proposals.
* Updates `antd@3.13.0` to get some `Affix` bug fixes they've made.

### Steps to Test

* Create an RFP and browse to it, confirm it lists no proposals at the bottom and it shows an affixed create button
* Click the create button, confirm a linked proposal is made for you
* Change a field, go back to the RFP and click create again, confirm you were taken back to your existing proposal rather than creating a new one
* Fill out the RFP and preview it, confirm it looks good
* Submit the proposal, confirm it still doesn't show up on RFP page
* Go to the admin page for the RFP, confirm your proposal shows up in the pending list
* Go to the admin page for the proposal and accept the proposal, confirm it now shows up in the "accepted proposals" section on the RFP admin page
* Publish the proposal on the pubic site, confirm it shows up on the RFP page now

## Screenshots

![screen shot 2019-01-30 at 11 48 53 pm](https://user-images.githubusercontent.com/649992/52033558-6748fc00-24f3-11e9-8f6b-f57e4e275c39.png)

![screen shot 2019-01-30 at 11 48 49 pm](https://user-images.githubusercontent.com/649992/52033559-6748fc00-24f3-11e9-9046-c8fd0ee2e026.png)

<img width="1532" alt="screen shot 2019-01-30 at 11 48 02 pm" src="https://user-images.githubusercontent.com/649992/52033561-6748fc00-24f3-11e9-9bec-e1b21ae26cc0.png">

<img width="1532" alt="screen shot 2019-01-31 at 12 36 36 am" src="https://user-images.githubusercontent.com/649992/52033557-6748fc00-24f3-11e9-8d04-d2de2dd1ccd5.png">
